fili-core/src/test/groovy/com/yahoo/bard/webservice/table/LogicalTableSchemaSpec.groovy

```diff
@@ -0,0 +1,39 @@
+package com.yahoo.bard.webservice.table
+
+import com.yahoo.bard.webservice.data.config.names.ApiMetricName
+import com.yahoo.bard.webservice.data.metric.LogicalMetric
+import com.yahoo.bard.webservice.data.metric.LogicalMetricColumn
+import com.yahoo.bard.webservice.data.metric.MetricDictionary
+import com.yahoo.bard.webservice.data.time.DefaultTimeGrain
+
+import spock.lang.Specification
+
+class LogicalTableSchemaSpec extends Specification {
+
+    def "Build metric called the apiMetricName is Valid and filters metrics as indicated" () {
+        String metricName1 = "name1"
+        String metricName2 = "name2"
+
+        LogicalMetric logicalMetric1 = Mock(LogicalMetric)
+        LogicalMetric logicalMetric2 = Mock(LogicalMetric)
+
+        ApiMetricName apiName1 = Mock(ApiMetricName) {
+            getApiName() >> metricName1
+        }
+        ApiMetricName apiName2 = Mock(ApiMetricName) {
+            getApiName() >> metricName2
+        }
+        1 * apiName1.isValidFor(logicalMetric1, _) >> true
+        1 * apiName2.isValidFor(logicalMetric2, _) >> false
```

Comment:
This is a small thing, but could we give these better names that "1" and "2"? Perhaps something like "validMetric" and "invalidMetric" so that it was easier to connect the dots about what's being tested, and why these mock values make sense?